### ISSUE TITLE
[Spike] - Research how appt notifications are triggered and mapped to a questionnaire

### ISSUE THREAD

**Issue by abuckley3385** [department-of-veterans-affairs/va.gov-team#13145]
Delighter for sprint 30 

As a Veteran, I use MHV and VA.gov to access all my health and benefits information with the VA, and want to be able to access my health questionnaires in a similar way, so the workflow is consistent and convenient to my current experience.

VAOS/CPRS - association of facility/encounter to trigger questionnaire that applies which will provide the ability to add a link to the notification the questionnaire is available.

- [ ] Our questionnaire service needs to know when to send the veteran a questionnaire, therefore we need the ability to know when a given appt. type at a facility is scheduled and we have mapped that to a specific questionnaire to be sent.  
- [ ] Need to identify where in CPRS we are listening to the appts that get booked
- [ ] Need to build a mapping of Appt. type and facility to clipboard (we will have multiple eventually)

## Questions to be answered
- [ ] What are the business requirements needed to satisfy the MVP?
- [ ] What are the technology requirements for the feature?

**Comment by massrb** [comment]
https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/discovery/engineering/trigger.md

**Comment by abuckley3385** [comment]
@massrb this ticket is not about how and what solution we are using to notify the veteran of a questionnaire - the Questionnaire trigger is to determine where in the source systems we are being notified an appt. was booked and how that is mapped to our specific questionnaire.  Please se the user stories above and acceptance descriptions.  I would suggest moving your document on vet text to a new ticket and associate it to the EPIC - Feature for Notifications

**Comment by mdewey** [comment]
The trigger that is being defined here is: When a user makes an appointment of type X, at facility Y, we know that they have a form of Z

**Comment by massrb** [comment]
The diagram was about triggers but was confusing because it suggested using an SMS text message which was an idea I talked about with Mark for triggers. The diagram will need to be changed and has been marked as such for now

**Comment by massrb** [comment]
We want to make sure that there aren't other options besides looking to VeText to get notified when an appointment is booked. Can we poll something for new appointments ? 

Recent discussion:

Question: Could we look at VeText code and figure out what they are doing and see if we can do the same thing ? I guess I thought they have hooks into the system that we couldn't have in the same manner but this question came up in a discussion

Stephen Barrs  10:39 AM
VetText has a custom Weblogic based implementation of VistaLink (I think thats the one) that lets them integrate with all Vistas. we def dont want to stand up our own VistALink

**Comment by abuckley3385** [comment]
@massrb  per your Slackup yesterday -  should this ticket be in progress?
>  I updated the trigger document

**Comment by massrb** [comment]
These are the meeting notes from last Friday with Shane Elliot 

https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/discovery/engineering/vetext-meeting-sept18.md

**Comment by massrb** [comment]
Current comments from changes to meeting notes from myself:

* It was stressed that we do not want to store appointment data in a clipboard DB, but there is fhir based questionnaire response type data in the PGD. (what does this have impact with storing data if we do or don't use FHIR??)*

> I thought that we are using FHIR

* Appointment IDs may be synthetic and outside of vista specifically (what impact does this have??)

> I am not totally sure, but it is an important detail I didn’t want to forget 

* A template variable associated with then clipboard Questionnaire could be added. What does this get us/why do we need this?

> I asked Shane that in a slack message today

Is the questionnaire admin tool is part of the Health Care Experience Clipboard or VeText

**Comment by massrb** [comment]
I asked Shane a question this morning at 11:00 on a channel with Mark and Stephen about the template variable. At the end of the day responded that he was going to draw something up

**Comment by massrb** [comment]
Shane presented a few options which we had been thinking of option 1 unless the simpler option 3 is suitable 

https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/discovery/engineering/assets/clipboard-VEText.jpg

**Comment by massrb** [comment]
I updated the document to include an illustration of a programatic flow

https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/discovery/engineering/vetext-meeting-sept18.md

**Comment by massrb** [comment]
What is a stop code ? 

Hi @Shane Elliott, I understand that the stop code will be potentially associated with a questionnaire if not with a clinic. I was wondering what is a stop code ? I got this definition by searching the web for "veterans affairs stop codes" though the name sounds like something could be associated with something being stopped but maybe not ?

> What are Stop Codes? ... DSS Identifiers are 3 digit, standardized codes used to characterize VHA outpatient clinics. They identify the work group primarily responsible for providing a clinical service, and are used for purposes of workload credit, managerial accounting, and program evaluation.

Shane:
>that is the correct definition. Think of it as a clinic category, cardiology, primary care etc

**Comment by massrb** [comment]
Added diagram: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/discovery/engineering/vetext-meeting-sept18.md#vets-apivetext-flow

**Comment by massrb** [comment]
I realized there is another big question so I am looking into this and asking questions:

> We are planning to expose an API to a VA related app called VeText. We want VeText to be able to hit our API, however it seems like vets-api is set up so that all access is through authenticated users to hit any endpoint. Is there any way to do this using a key or any other option ?

**Comment by mdewey** [comment]
Start this: still WIP and needs nonengineering team buy-in: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/engineering/questionnaire-trigger/0.mvp.md

**Comment by abuckley3385** [comment]
@mdewey and @dillo  I really like the simplicity and not over engineering our POC.  Lets get time to discuss as a group and ensure we can identify issues now to work through.  Maybe posting on questionnaire tech questions thread we started and add a few of the VA engineers for thoughts.  I think we can address your questions in the bottom easily.

**Comment by mdewey** [comment]
Yes! I'm not quite done with all the docs and research but I will schedule
something when I get a little further into it from an engineering standpoint

**Comment by mdewey** [comment]
Meeting scheduled for Tuesday, Nov 3rd. 2020

**Comment by mdewey** [comment]
Asked questions to tech team : 
https://dsva.slack.com/archives/C019EATUKN2/p1604516347013500

**Comment by mdewey** [comment]
Next steps, get questions answered and create tickets based on action items

**Comment by mdewey** [comment]
Waiting on meeting with MAP folks to discuss FHIR spec questions.

**Comment by mdewey** [comment]
Set a meeting for Thursday, Dec 17th for the engineering team to work on this.

**Comment by mdewey** [comment]
Enginneering team met

Notes:

**Comment by mdewey** [comment]
Current Notes: 
https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/engineering/questionnaire-trigger/meeting%20notes/2021.01.11.md

**Comment by mdewey** [comment]
Latest notes: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/engineering/questionnaire-trigger/meeting%20notes/2012.01.13.md

Meeting on Friday (01/15/2021) with the team to share findings

**Comment by mdewey** [comment]
Summary: 
There will be no push trigger. We will be creating a service that will allow other apps to query if an appointment. 

Here is the summary : https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/health-care/questionnaire/engineering/questionnaire-trigger
